AndroidStudioProjects/BMIApp/app/src/main/java/com/example/bmiapp/ui/input/InputFragment.kt

```diff
@@ -28,26 +33,33 @@ class InputFragment : Fragment() {
     //bmi
     private var bmi: String = ""
     //sharedのkeyリスト
-    private lateinit var keyList: MutableList<String>
+    private var keyList: MutableList<String>? = null
     //同日の計測データの存在判定真理値
     private var isFirst = true
     //保存日の日付
     private var strToday = ""
     //入力値の正誤判定真理値
     private var isCorrectValue = false
-    private lateinit var res: Resources
-    private lateinit var sharedPreferences: SharedPreferences
-    private lateinit var editor: SharedPreferences.Editor
+    private var res: Resources? = null
+    private var sharedPreferences: SharedPreferences? = null
+    private var editor: SharedPreferences.Editor? = null
+
+    companion object {
+        const val DATE = "日付"
+        const val HEIGHT = "身長"
+        const val WEIGHT = "体重"
+        const val BMI = "BMI"
+        const val MESSAGE = "メッセージ"
+        const val ISFIRST = "初回判定"
+    }
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(requireContext())
+        sharedPreferences = getDefaultSharedPreferences(requireContext())
```

Comment:
ここが、deprecationの場所であれば、以下のような形にするおtよいと思います
```suggestion
    sharedPreferences = context?.getSharedPreferences(context!!.packageName, Context.MODE_PRIVATE)
```